filesystems/nfs/connectathon/runtest.sh

```diff
@@ -0,0 +1,561 @@
+       _types=`echo $_types | sed -r "s/(tcp|udp)6//g"`
+    fi
+
+    echo "Supported IP types for $server/$nfsver: $_types" | tee -a $OUTPUTFILE
+    eval "$4='$_types'"
+}
+
+function cthon_all ()
+{
+    local server=$1
+    local nfspath=$2
+    local ipv6=$3
+    local SCORE=0
+    local rc=0
+    # globals:
+    #   client_nfsvers
+    #   server_nfsvers
+    #   types
+
+    get_supported_client_versions client_nfsvers
+    get_supported_server_versions "$server" server_nfsvers
+    [[ -n "$NFS_VERS" ]] && client_nfsvers=$NFS_VERS
+
+    for nfsver in $client_nfsvers; do
+        # check if server supports this nfsver
+        echo "$server_nfsvers" | grep "$nfsver" > /dev/null
+        if [ -z "$NFS_VERS" -a $? -ne 0 ]; then
+            echo "$server does not support NFS:$nfsver"
+            continue
+        fi
+
+        get_ip_types "$server" "$nfsver" $ipv6 types
+        for type in $types; do
+            for test_name in $tests; do
+                echo "cthon_all: NFS v$nfsver IP:$type TEST:$test_name" | tee -a $OUTPUTFILE
+                case $nfsver in
+                [2-3])
+                        # _cthon04 fstype test_name report_path server nfspath options
+                        _cthon04 nfs $test_name "nfsvers=${nfsver}_${type}" $server $nfspath -onfsvers=$nfsver,proto=$type
+                        rc=$?
+                        ;;
+                4)
+                        # _cthon04 $fstype $test_name $report_path $server $nfspath $options
+                        _cthon04 nfs4 $test_name "nfsvers=${nfsver}_${type}" $server $nfspath -oproto=$type,$MOUNT_OPT
+                        rc=$?
+                        ;;
+                pNFS)
+                        _cthon04 nfs4 $test_name "pNFS_${type}" $server / -ominorversion=1,proto=$type,$MOUNT_OPT
+                        rc=$?
+                        ;;
+                esac
+                [ $rc -ne 0 ] && SCORE=$(expr $SCORE + 1)
+                echo -e "nfsvers=$nfsver\t$type\t$test_name\t$rc" >> result.txt
+            done
+        done
+    done
+
+    return $SCORE
+}
+
+
+function cthon_main ()
+{
+    pushd cthon04
+    for server_path in $servers; do
+        : > result.txt
+
+        # start time
+        local STIME=`date +%s`
+        local server=$(echo $server_path| cut -f1 -d:)
+        local nfspath=$(echo $server_path| cut -f2 -d:)
+        local SCORE=0
+        local server_is_pnfs=0
+        local now=`date`
+        local ipv6=0
+
+        #
+        # ServerStatus: Lets see if the server is online
+        #
+        ping -c 3 $server >> $OUTPUTFILE 2>&1
+        if [ $? -ne 0 ]; then
+            echo "" | tee -a $OUTPUTFILE
+            echo " ===== cthon_main: $server is offline =====" | tee -a $OUTPUTFILE
+            echo " ***** INFO: $now *****" | tee -a $OUTPUTFILE
+            echo "" | tee -a $OUTPUTFILE
+            report_result $server_IPv4_Status_offline WARN/COMPLETED
```

Comment:
@veruu, The 'server' at L450 and L464 are from 'server_path' in a 'for' loop, 
if one 'server' doesn't satisfy the test requirements (e.g. ping, ping6),
it does try to skip that server (L457) and go to pick up the next one, do we
have to had a dependency on the first server in $servers?

+++ code segment +++
   432	function cthon_main ()
   433	{
   434	    pushd cthon04
   435	    for server_path in $servers; do
   440	        local server=$(echo $server_path| cut -f1 -d:)
   ...
   450	        ping -c 3 $server >> $OUTPUTFILE 2>&1
   451	        if [ $? -ne 0 ]; then
   ...
   456	            report_result $server_IPv4_Status_offline WARN/COMPLETED
   457	            continue
   458	        fi
   ...
   462	        if rpcinfo -s $server | grep 100003 | grep tcp6; then
   463	            ipv6=1
   464	            ping6 -c 3 $server >> $OUTPUTFILE 2>&1
   465	            if [ $? -ne 0 ]; then
   ...
   472	                report_result $server_IPv6_Status_offline WARN/COMPLETED
   473	            fi
   474	        fi
   475	
   476	        echo "cthon_main, $server is online" | tee -a $OUTPUTFILE
   477	
   478	        echo "Running cthon_all" | tee -a $OUTPUTFILE
   479	        cthon_all $server $nfspath $ipv6
   ...
   503	    done
   504	    popd
   505	}
+++ END +++

Reply:
I'm not sure what you mean by "it does try to skip that server (L457) and go to pick up the next one", can you elaborate? The `report_result WARN/COMPLETED` suggests it reports offline servers as a testing failure.

Reply:
@idorax all the servers in the log you posted are online so that doesn't help. I'm asking, when will the `report_result $server_IPv4_Status_offline WARN/COMPLETED` be run?

Reply:
@idorax all the servers in the log you posted are online so that doesn't help. I'm asking, when will the `report_result $server_IPv4_Status_offline WARN/COMPLETED` be run?

Reply:
@idorax yes it does

Reply:
Since function checkServers() has helped to filter those offline servers out before the real tests are started, I'd like to simply skip the test run if a server is not pingable. To make you review easily, I also created an incremental commit, https://github.com/idorax/tests-beaker/commit/42f37d3c4b0373bf9780a52920eef1872dc7ac75. Please take a look again, thanks!